kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -347,6 +454,23 @@ fn parse_path(meta_item: &MetaItem) -> Option<String> {
     }
 }
 
+/// Parse the arguments of the attribute into a (key, value) map.
+fn parse_key_values(attr: &Attribute) -> Result<BTreeMap<String, String>, String> {
+    trace!(list=?attr.meta_item_list(), ?attr, "parse_key_values");
+    let args = attr.meta_item_list().ok_or("malformed attribute input")?;
+    args.iter()
+        .map(|arg| match arg.meta_item() {
+            Some(MetaItem { path: key, kind: MetaItemKind::NameValue(val), .. }) => {
+                Ok((key.segments.first().unwrap().ident.to_string(), val.symbol.to_string()))
+            }
+            _ => Err(format!(
+                r#"expected "key = value" pair, but found `{}`"#,
```

Comment:
Does Rust reports these errors the same way?

Reply:
No, but I don't think it's worth to walk the AST just for this. I don't think the error that rustc emits is that much more friendly. Here are a few examples:
```
error: expected unsuffixed literal or identifier, found `<eof>`
 --> src/lib.rs:5:12
  |
5 | #[cfg(test =)]
  |            ^
```
or
```
error: expected one of `(`, `,`, `::`, or `=`, found `?`
 --> src/lib.rs:5:12
  |
5 | #[cfg(test ? aff)]
  |            ^ expected one of `(`, `,`, `::`, or `=`

```

Reply:
Agree, thanks for checking!